lib/main.coffee

```diff
@@ -99,17 +99,18 @@ module.exports = Hydrogen =
         language = command.language
         grammar = command.grammar
         kernelInfo = command.kernelInfo
+        kernel = KernelManager.getRunningKernelForLanguage language
 
         if request is 'interrupt-kernel'
-            KernelManager.interruptKernelForLanguage language
+            kernel?.interrupt()
 
         else if request is 'restart-kernel'
-            KernelManager.destroyKernelForLanguage language
+            KernelManager.destroyRunningKernelForLanguage language
             @clearResultBubbles()
             KernelManager.startKernelIfNeeded language
 
         else if request is 'switch-kernel'
-            KernelManager.destroyKernelForLanguage language
+            KernelManager.destroyRunningKernelForLanguage language
```

Comment:
@rgbkrk The naming in this class is something we need to revisit. I added `Running` to hint that this method must be invoked. If a kernel is destroyed by calling `kernel.destroy()` directly, 'KernelManager` won't be aware of it and will keep it listed amongst its running kernels.